src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilOtherCompleted() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnCompleted();
+
+        result.assertError(CancellationException.class);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+
+    }
+
+    private static class TestObservable implements Observable.OnSubscribe<String> {
+
+        Observer<? super String> observer = null;
+        Subscription s;
+
+        public TestObservable(Subscription s) {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnCompleted() {
+            observer.onCompleted();
+        }
+
+        /* used to simulate subscription */
+        public void sendOnNext(String value) {
+            observer.onNext(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            observer.onError(e);
+        }
+
+        @Override
+        public void call(Subscriber<? super String> observer) {
+            this.observer = observer;
+            observer.add(s);
+        }
+    }
+
+    private static class TestSingle implements Single.OnSubscribe<String> {
+
+        SingleSubscriber<? super String> subscriber = null;
+        Subscription s;
+
+        public TestSingle(Subscription s) {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnSuccess(String value) {
+            subscriber.onSuccess(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            subscriber.onError(e);
+        }
+
+        @Override
+        public void call(SingleSubscriber<? super String> observer) {
+            this.subscriber = observer;
+            observer.add(s);
+        }
+    }
+
+    @Test
+    public void testUntilFires() {
+        PublishSubject<Integer> source = PublishSubject.create();
+        PublishSubject<Integer> until = PublishSubject.create();
+
+        TestSubscriber<Integer> ts = new TestSubscriber<Integer>();
+
+        source.take(1).toSingle().takeUntil(until.take(1).toSingle()).unsafeSubscribe(ts);
+
+        assertTrue(source.hasObservers());
+        assertTrue(until.hasObservers());
+
+        until.onNext(1);
+
+        ts.assertError(CancellationException.class);
+        ts.assertTerminalEvent();
```

Comment:
Not needed since you already checked `ts.assertError` but you can check `ts.assertNotCompleted()` instead (not sure it exists though, reviewing from phone)